### PR TITLE
Add log-level config option

### PR DESCRIPTION
**Pull request by laurenceisla**
Added documentation on the `log-level` config option.

### CODE REVIEW

File: configuration.rst [hunk 1]
```diff
@@ -176,6 +177,27 @@ server-unix-socket-mode
...
+      # No requests are logged, only startup and db connection recovery messages are.
+      log-level = "crit"
+
+      # All the "crit" level events plus server errors (status >= 500) are logged
+      log-level = "error"
+
+      # All the "error" level events plus request errors (status >= 400) are logged
+      log-level = "warning"
+
+      # All requests are logged
```
Comment (steve-chavez) [inline configuration.rst:198]: Clarifying verbosity:
```suggestion
      # All the "warning" level events plus all requests(every status code) are logged
```

File: releases/upcoming.rst [hunk 2]
```diff
@@ -19,6 +19,9 @@ Added
 * Config options for showing a full OpenAPI output regardless of the JWT role privileges and for disabling it altogether. See :ref:`openapi-mode`.
   |br| -- `@steve-chavez <https://github.com/steve-chavez>`_
 
+* Config option for selecting the logging verbosity level. See :ref:`log-level`.
```
Comment (steve-chavez) [inline releases/upcoming.rst:22]:
```suggestion
* Config option for logging level. See :ref:`log-level`.
```

**Comment by steve-chavez** [comment]
@laurenceisla We also need a Changed entry in [upcoming](https://postgrest--414.org.readthedocs.build/en/414/releases/upcoming.html#changed), the log-level used to be `info` by default.

File: configuration.rst [hunk 3]
```diff
@@ -176,12 +177,33 @@ server-unix-socket-mode
...
+.. _log-level:
+
+log-level
+---------
+
+  Specifies the level of information to be logged while running PostgREST.
+
+  .. code:: bash
+
+      # No requests are logged, only startup and db connection recovery messages are.
```
Comment (steve-chavez) [inline configuration.rst:189]:
```suggestion
      # Only startup and db connection recovery messages are logged, requests are not logged.
```

Comment (laurenceisla) [reply configuration.rst:189]: Maybe, not mention the requests at all and leave it like this?

```suggestion
      # Only startup and db connection recovery messages are logged
```

File: configuration.rst [hunk 4]
```diff
@@ -176,12 +177,33 @@ server-unix-socket-mode
...
+      log-level = "crit"
+
+      # All the "crit" level events plus server errors (status >= 500) are logged
```
Comment (laurenceisla) [inline configuration.rst:192]: I'm not sure if the `>=` usage is confusing. Maybe, it should be replaced for the status class?

```suggestion
      # All the "crit" level events plus server errors (status 5xx) are logged
```

File: configuration.rst [hunk 5]
```diff
@@ -176,12 +177,33 @@ server-unix-socket-mode
...
+      log-level = "error"
+
+      # All the "error" level events plus request errors (status >= 400) are logged
```
Comment (laurenceisla) [inline configuration.rst:195]: Ditto.

```suggestion
      # All the "error" level events plus request errors (status 4xx) are logged
```

File: configuration.rst (hunk 3 above)
Comment (steve-chavez) [reply configuration.rst:189]: Yeah, agree.

**Pull request merged**